frontend/vite.config.ts

```diff
@@ -1,15 +1,28 @@
 import { URL, fileURLToPath } from 'node:url';
+import { defineConfig, loadEnv } from 'vite';
 
-import { defineConfig } from 'vite';
 import vue from '@vitejs/plugin-vue';
 import vueJsx from '@vitejs/plugin-vue-jsx';
 
 // https://vitejs.dev/config/
-export default defineConfig({
-    plugins: [vue(), vueJsx()],
-    resolve: {
-        alias: {
-            '@': fileURLToPath(new URL('./src', import.meta.url)),
+export default defineConfig(({ mode }) => {
+    const env = loadEnv(mode, process.cwd());
+    const PORT = env.VITE_QUIZOOT_SERVER_PORT ?? '8000';
+
+    return {
+        plugins: [vue(), vueJsx()],
+        resolve: {
+            alias: {
+                '@': fileURLToPath(new URL('./src', import.meta.url)),
+            },
+        },
+        server: {
+            proxy: {
```

Comment:
every requests should starts with `/api`. 

`fetch('/api/quizzes')` will be redirected to `http://localhost:8000/api/quizzes`.

Reply:
Can't we do it like this ?
```typescript
proxy: {
    '/': {
        target: `http://localhost:${PORT}/api`,
        changeOrigin: true,
     },
}
```
So as `fetch('/quizzes')` would be redirected to `http://localhost:8000/api/quizzes`

Reply:
No ! Because we want to route frontend trafic to backend.

`fetch('/api/quizzes')` is a request with the full path `http://localhost:FRONTEND_PORT/api/quizzes` and that will be proxied to `http://localhost:BACKEND_PORT/api/quizzes`.

We cannot use `/` because frontend will route `http://localhost:FRONTEND_PORT/quizzes` to `http://localhost:BACKEND_PORT/api/quizzes`. But you know that `http://localhost:FRONTEND_PORT/quizzes` is the VIEW that list all the quizzes...  you don't want to re-route that


Basically, we only route traffic from `FRONTEND/api` to `BACKEND/api` since frontend doesn't implement the services.

Reply:
Oh I see now. A big misundersting of how the proxy actually works: `Any requests that request path starts with that key will be proxied to that specified target` (from [Vite Doc](https://vitejs.dev/config/server-options.html#server-proxy)).
Thanks !